src/primitives/hrp.rs

```diff
@@ -148,6 +148,14 @@ impl Hrp {
     #[inline]
     pub fn to_lowercase(&self) -> String { self.lowercase_char_iter().collect() }
 
+    /// Returns this human-readable part as bytes.
+    #[inline]
+    pub fn as_bytes(&self) -> &[u8] { &self.buf }
+
+    /// Returns this human-readable part as str.
+    #[inline]
+    pub fn as_str(&self) -> Result<&str, str::Utf8Error> { str::from_utf8(&self.buf) }
```

Comment:
We can use expect here because we only store ASCII bytes, then there is no need to return a `Result`.